### DISCUSSION TITLE
request.json can only be consumed once

### DISCUSSION THREAD

**Discussion by xoewrl** [fastapi/fastapi#9622] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, Request
import uvicorn


app = FastAPI()


@app.middleware("http")
async def middle_set(request: Request, call_next):
    body = await request.json()
    print(body)
    response = await call_next(request)
    return response

@app.post("/")
async def root(request: Request):
    body = await request.json()
    return {"got parmas": body}

if __name__ == '__main__':
    module_name = __file__.split("/")[-1].split("\\")[-1].replace(".py", "")
    uvicorn.run(f'{module_name}:app',host="0.0.0.0", port=9000,reload=True)
```

### Description

If  **await requetst.json()**  inside a **middleware**,
Then the **endpoint handler function**（root function）cannot get value from **await request.json()**. 

It will **stuck** (if request headers is keep-alive)   or 
raise （if request headers is connection:close） 
**"anyio.EndOfStream"** 

### Operating System

Linux

### FastAPI Version

0.95.1

### Python Version

3.8.0

**Comment by Kludex** [comment, marked as answer]
This is a Starlette issue, which is going to be solved in the next Starlette release.

**Reply by xoewrl** [reply]
I see, thank you !

**Reply by fredi-python** [reply]
Any update to this?
(Nevermind, just updated my starlette library)

**Comment by Kludex** [comment]
Locking the discussion to avoid confusion. This was already solved.
